### PR TITLE
ARROW-7749: [C++] Link more tests together

### PR DESCRIPTION
**Pull request by pitrou**
This should make unity builds slightly faster, by allowing CMake to coalesce
more source files and save on header file parsing.

### CODE REVIEW

File: cpp/src/arrow/CMakeLists.txt [hunk 1]
```diff
@@ -514,8 +516,13 @@ if(ARROW_IPC)
   add_arrow_test(extension_type_test)
 endif()
 
-add_arrow_test(memory_pool_test)
-add_arrow_test(pretty_print_test)
+add_arrow_test(misc_test
```
Comment (bkietz) [inline cpp/src/arrow/CMakeLists.txt]:
```suggestion
add_arrow_test(core_types_test
```
?

**Review by bkietz — APPROVED** [review]

File: cpp/src/arrow/CMakeLists.txt (hunk 1 above)
Comment (pitrou) [reply cpp/src/arrow/CMakeLists.txt]: "Core types" sounds like "datatypes" to me, so it looks a bit misleading.

Comment (pitrou) [reply cpp/src/arrow/CMakeLists.txt]: I agree "misc" isn't satisfactory :-)

**Review by wesm — APPROVED** [review]
+1

**Pull request closed without merging**